src/dialect/mod.rs

```diff
@@ -300,13 +304,172 @@ pub trait Dialect: Debug + Any {
         // return None to fall back to the default behavior
         None
     }
+
+    /// Get the precedence of the next token. This "full" method means all precedence logic and remain
```

Comment:
This is moved to `Dialect` so dialects can change the precedence without all the numeric constants being shared between modules.

it's implemented as a separate function from `get_next_precedence`:
1. To avoid breaking the API
2. To allow dialects to customise precedence but optionally fall back to the default behaviour, as demonstrated `Snowflake`.